x/ibc/04-channel/keeper/packet.go

```diff
@@ -326,6 +329,43 @@ func (k Keeper) ReceiveExecuted(
 	return nil
 }
 
+// WriteAcknowledgement writes the packet execution acknowledgement to the state,
+// which will be verified by the counterparty chain using AcknowledgePacket.
+//
+// CONTRACT: this function must be called in the IBC handler
```

Comment:
As @colin-axner described, this function doesn't need to be called in the IBC handler.

Reply:
No, although it needs to be called by the module which originally processed the packet